Objects/Converters/ConverterRevit/ConverterRevitShared/Partial Classes/ConvertFamilyInstance.cs

```diff
@@ -70,8 +81,94 @@ public ApplicationObject FamilyInstanceToNative(BuiltElements.Revit.FamilyInstan
+          }
+          else if (familySymbol.Family.FamilyPlacementType == FamilyPlacementType.WorkPlaneBased)
+          {
+            if (CurrentHostElement == null)
+              Report.ConversionErrors.Add(new Exception($"Object with ID {speckleFi.id} is work plane based, but does not have a host element"));
+            if (CurrentHostElement is Wall wall)
+            {
+              Doc.Regenerate();
+
+              Options op = new Options();
+              op.ComputeReferences = true;
+              GeometryElement wallGeom = wall.get_Geometry(op);
+              Reference faceRef = null;
+
+              foreach (var geom in wallGeom)
+              {
+                if (geom is Solid solid)
+                {
+                  FaceArray faceArray = solid.Faces;
+
+                  double planeDist = double.PositiveInfinity;
+                  foreach (Face face in faceArray)
+                  {
+                    if (face is PlanarFace planarFace)
+                    {
+                      // some family instance base points may lie on the intersection of faces
+                      // this makes it so family instance families can only be placed on the
+                      // faces of walls
+                      if (NormalsAlign(planarFace.FaceNormal, wall.Orientation))
+                      {
+                        double newPlaneDist = ComputePlaneDistance(planarFace.Origin, planarFace.FaceNormal, basePoint);
+                        if (newPlaneDist < planeDist)
+                        {
+                          planeDist = newPlaneDist;
+                          faceRef = planarFace.Reference;
+                        }
+                      }
+                    }
+                  }
+                }
+              }
+
+              // last resort, just guess a face
+              if (faceRef == null)
+                faceRef = HostObjectUtils.GetSideFaces(wall, ShellLayerType.Interior)[0];
+
+              XYZ norm = new XYZ(0, 0, 0);
+              familyInstance = Doc.Create.NewFamilyInstance(faceRef, basePoint, norm, familySymbol);
+
+              // parameters
+              IList<Parameter> cutVoidsParams = familySymbol.Family.GetParameters("Cut with Voids When Loaded");
+              IList<Parameter> lvlParams = familyInstance.GetParameters("Schedule Level");
+
+              if (cutVoidsParams.ElementAtOrDefault(0) != null && cutVoidsParams[0].AsInteger() == 1)
+                InstanceVoidCutUtils.AddInstanceVoidCut(Doc, wall, familyInstance);
+              if (lvlParams.ElementAtOrDefault(0) != null)
+                lvlParams[0].Set(level.Id);
+            }
+            else if (CurrentHostElement is Floor floor)
+            {
+              // TODO: support hosted elements on floors. Should be very similar to above implementation
+              Report.ConversionErrors.Add(new Exception($"Work Plane based families on floors to be supported soon"));
```

Comment:
Logging should be updated to use the new applicationobject

Reply:
I created a separate issue for this so we don't forget! :D Good catch!

Reply:
https://github.com/specklesystems/speckle-sharp/issues/1473